modules/account-lib/src/coin/stx/contractBuilder.ts

```diff
@@ -0,0 +1,131 @@
+  private _options: UnsignedContractCallOptions | UnsignedMultiSigContractCallOptions;
+  private _contractAddress: string;
+  private _contractName: string;
+  private _functionName: string;
+  private _functionArgs: ClarityValue[];
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  initBuilder(tx: Transaction): void {
+    const txData = tx.toJson();
+    if (txData.payload === undefined) {
+      throw new InvalidTransactionError('payload must not be undefined');
+    }
+    if (txData.payload.payloadType == PayloadType.ContractCall) {
+      this.contractAddress(txData.payload.contractAddress)
+      this.contractName(txData.payload.contractName)
+      this.functionName(txData.payload.functionName)
+      this.functionArgs(txData.payload.functionArgs)
+      super.initBuilder(tx);
+    } else {
+      throw new BuildTransactionError("Transaction should be contract call")
+    }
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._options = this.buildContractCallOptions();
+    this.transaction.setTransactionType(TransactionType.ContractCall);
+    this.transaction.stxTransaction = await makeUnsignedContractCall(this._options);
+    return await super.buildImplementation();
+  }
+
+  private buildContractCallOptions(): UnsignedContractCallOptions | UnsignedMultiSigContractCallOptions {
+    const defaultOpts: ContractCallOptions = {
+      contractAddress: this._contractAddress,
+      contractName: this._contractName,
+      functionName: this._functionName,
+      functionArgs: this._functionArgs,
+      network: this._network,
+      fee: new BigNum(this._fee.fee),
+      nonce: new BigNum(this._nonce),
+    };
+    if (this._multiSignerKeyPairs.length === 0) {
+      throw new InvalidParameterValueError('supply at least 1 public key');
+    }
+    if (this._multiSignerKeyPairs.length === 1) {
+      return {
+        ...defaultOpts,
+        publicKey: this._multiSignerKeyPairs[0].getKeys(this._multiSignerKeyPairs[0].getCompressed()).pub,
+      };
+    } else {
+      return {
+        ...defaultOpts,
+        publicKeys: this._multiSignerKeyPairs.map(k => k.getKeys(k.getCompressed()).pub),
+        numSignatures: this._numberSignatures,
+      };
+    }
+  }
+
+  //region Contract fields
+  /**
+   * Set the contract address
+   *
+   * @param {string} address the address deployed the contract
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  contractAddress(address: string): this {
+    if (!isValidAddress(address)) {
+      throw new InvalidParameterValueError('Invalid address');
+    }
+    this._contractAddress = address;
+    return this;
+  }
+
+  /**
+   * Set the contract name
+   *
+   * @param {string} name name of contract
+   * @returns {TransferBuilder} the builder with the new parameter set
+   */
+  contractName(name: string): this {
+    if (name.length == 0) {
```

Comment:
Maybe also a question for Tyler: can we restrict the contract name to just https://explorer.stacks.co/txid/SP000000000000000000002Q6VF78.pox (pox) for mainnet and ST000000000000000000002AMW42H.pox (pox) for testnet here or should we do it in platform? 

Purpose is that we will only support interactions with the pox contract for blockstack for this project

Reply:
Sure, I don't see a problem with that if we really ever only want to interact with these contracts - better to fail as early as possible. It does keep the library from working for other use cases, but we can always remove/adjust this check if needed later.